backend/src/swa-tracker/swa-tracker-wrapper.ts

```diff
@@ -20,89 +20,97 @@ const EVENT_TYPES = {
+			"SAPSE",
+			YEOMAN_UI,
+			// callback for error, one such callback for all the errors we receive via all the track methods
+			// error can be string (err.message) or number (response.statusCode)
+			(error: string | number) => {
+				if (typeof error === 'string') {
+					getLogger().error("SAP Web Analytics tracker failed to track", {errorMessage: error});
+				} else if (typeof error === 'number') {
+					// bug in matomo-tracker: they think that success is 200 or 30[12478], so we are rechecking here
+					if (!((error >= 200) && (error <= 299)) && !((error >= 300) && (error <= 399))) {
+						getLogger().error("SAP Web Analytics tracker failed to track", {statusCode: error});
+					}
 				}
 			}
+		  );
+	
+		// Update the swa-tracker-wrapper with a reference to the swaTracker.
+		this.initSWATracker(swaTracker);
+		getLogger().info(`SAP Web Analytics tracker was created for ${YEOMAN_UI}`);
+	}
+	
+	
+	public static updateGeneratorStarted(generatorName: string) {
+		const eventType = EVENT_TYPES.PROJECT_GENERATION_STARTED;
+		let customEvents = [generatorName];
+		let startTime = this.startTime = Date.now();
+		this.getSWA().track(eventType, customEvents);
+		getLogger().trace("SAP Web Analytics tracker was called and start time was initialized", {
+			eventType, generatorName, startTime, customEvents});
+	}
+	
+	
+	public static updateGeneratorEnded(generatorName: string, isSucceeded: boolean, errorMessage?: string) {
+		let eventType = EVENT_TYPES.PROJECT_GENERATED_SUCCESSFULLY;
+		if (!isSucceeded) {
+			eventType = EVENT_TYPES.PROJECT_GENERATION_FAILED;
 		}
-	  );
-
-	// Update the swa-tracker-wrapper with a reference to the swaTracker.
-	initSWATracker(swaTracker);
-	getLogger().info(`SAP Web Analytics tracker was created for ${YEOMAN_UI}`);
-}
-
-export function updateGeneratorStarted(generatorName: string) {
-	const eventType = EVENT_TYPES.PROJECT_GENERATION_STARTED;
-	let customEvents = [generatorName];
-	startTime = Date.now();
-	getSWA().track(eventType, customEvents);
-	getLogger().trace("SAP Web Analytics tracker was called and start time was initialized", {
-		eventType, generatorName, startTime, customEvents});
-}
-
-export function updateGeneratorEnded(generatorName: string, isSucceeded: boolean, errorMessage?: string) {
-	let eventType = EVENT_TYPES.PROJECT_GENERATED_SUCCESSFULLY;
-	if (!isSucceeded) {
-		eventType = EVENT_TYPES.PROJECT_GENERATION_FAILED;
+	
+		const endTime = Date.now();
+		const startTime = this.startTime;
+		const generationTimeMilliSec = endTime - startTime;
+		const generationTimeSec = Math.round(generationTimeMilliSec/1000);
+		let customEvents = [generatorName, generationTimeSec.toString()];
+		if (!_.isNil(errorMessage)) {
+			customEvents.push(errorMessage);
+		}
+		this.getSWA().track(eventType, customEvents);
+		getLogger().trace("SAP Web Analytics tracker was called", 
+			{eventType, generatorName, generationTimeSec, generationTimeMilliSec, endTime, startTime, customEvents, errorMessage});
```

Comment:
We don't need local startTime variable.
Inside the json sent to logger: startTime: this.startTime